### ISSUE TITLE
Conditional Type Narrowing does not work with `in` operator

### ISSUE THREAD

**Issue by rayansostenes** [microsoft/pyright#3847]
**Describe the bug**
In the code snippet below the Literal type is narrowed correctly when using `==` operator but not when using the `in`

```python
from typing import Literal

from typing_extensions import assert_never, reveal_type

LettersLiteral = Literal[
    "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P"
]


def func1(letter: LettersLiteral) -> int:
    if (
        letter == "A"
        or letter == "B"
        or letter == "C"
        or letter == "D"
        or letter == "E"
        or letter == "F"
    ):
        return 1
    elif letter == "G" or letter == "H" or letter == "I" or letter == "J" or letter == "K":
        return 2
    elif letter == "L" or letter == "M" or letter == "N" or letter == "O" or letter == "P":
        return 3
    else:
        reveal_type(letter)
        assert_never(letter)


def func2(letter: LettersLiteral) -> int:
    if letter in ("A", "B", "C", "D", "E", "F"):
        return 1
    elif letter in ("G", "H", "I", "J", "K"):
        return 2
    elif letter in ("L", "M", "N", "O", "P"):
        return 3
    else:
        reveal_type(letter)
        assert_never(letter)
```

```
25:21 - information: Type of "letter" is "Never"
37:21 - information: Type of "letter" is "Literal['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P']"
38:22 - error: Argument of type "LettersLiteral" cannot be assigned to parameter "__arg" of type "NoReturn" in function "assert_never"
    Type "LettersLiteral" cannot be assigned to type "NoReturn"
      Type "Literal['A']" cannot be assigned to type "NoReturn" (reportGeneralTypeIssues)
```

Apparently, pyright can understand this in other contexts like `reportUnnecessaryContains`:
```python
def func3(letter: LettersLiteral) -> int:
    if letter in ("Q", "R", "S", "T", "U", "V"):
        return 5
    return 1
```
```
error: Expression will always evaluate to False since the types "Literal['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P']" and "Literal['Q', 'R', 'S', 'T', 'U', 'V']" have no overlap (reportUnnecessaryContains)
```

**Additional context**
pyright version: 1.1.267

**Comment by erictraut** [comment]
Pyright does partly handle type narrowing for the `in` operator when the RHS is a tuple of literal types. However, it currently handles this narrowing only in the positive ("if") case, not in the negative ("else") case. Interestingly, mypy does not handle this type narrowing form at all, even in the positive case.

I think it should be pretty straightforward to extend the narrowing support in the negative case.

**Comment by erictraut** [comment]
This will be included in the next release. 

Commit: https://github.com/microsoft/pyright/commit/b74be3b2cb2c5d35b960e8c6a06966d89b40c35a

**Comment by rayansostenes** [comment]
@erictraut Thanks!!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.268, which I just published. It will also be included in a future version of pylance.
